plugins/src/plugins/localization/gps.cpp

```diff
@@ -96,19 +113,32 @@ Gps::Reset()
 void
 Gps::send_position()
 {
-	if (gps_pub_->HasConnections()) {
+	if(gps_pub_->HasConnections() &&
+	   (!use_object_tracking_ || gps_world_pub_->HasConnections()))
+	{
```

Comment:
This means that the new pose is not published to `gps_pub_` if `gps_world_pub_` does not have a connection. Does that make sense?
Instead, you could also always create the `posMsg` and then only check if there is a connection when sending the message:
```
msgs::Pose posMsg;
// ...
if (gps_pub_->HasConnections()) { gps_pub_->Publish(posMsg); }
if (use_object_tracking_ && gps_world_pub_->HasConnection()) { gps_world_pub_->Publish(posMsg); }
```

Reply:
On a second thought, is the check for `use_object_tracking_` even necessary? I'd assume `gps_world_pub_->HasConnections()` would always be false if the node was not initialized. This would simplify the code, because you don't even need the `use_object_tracking_` member variable at all.

Reply:
Good point! However, use_object_tracking_  is necessary to decide if gps_world_pub_  should be created in the first place and in case it is not, gps_world_pub_->HasConnections() crashes gazebo when loading the plugin.

Reply:
And is there a way in cmake to format the code automatically like for fawkes-robotino with "make format"?